src/spikeinterface/sortingcomponents/benchmark/benchmark_matching.py

```diff
@@ -16,101 +17,483 @@
 
 import time
 import os
+from pathlib import Path
 import string, random
 import pylab as plt
+import matplotlib.patches as mpatches
 import numpy as np
+import pandas as pd
+import shutil
+import copy
+from tqdm.auto import tqdm
+
+
+def running_in_notebook():
+    try:
+        shell = get_ipython().__class__.__name__
+        notebook_shells = {"ZMQInteractiveShell", "TerminalInteractiveShell"}
+        # if a shell is missing from this set just check get_ipython().__class__.__name__ and add it to the set
+        return shell in notebook_shells
+    except NameError:
+        return False
```

Comment:
I do not like so much the idea that a piece of code is aware itself where it is run.
I think it is beeter to have clear flags to change a display behavior.

Reply:
What are the issues with this? I think it would be instructive for me if I knew exactly what you are concerned about.

Reply:
Lets leave this for now.
We will make a clean later.